Move check input

permcody opened the pull request:
So it seems to me that syntax check should terminate after successfully reading the input file and running the Actions. Prior to this PR we were also initializing the Executioner which would run `initialSetup` which does several more things:

 - projects ICs
 - executes all "execute_on = initial" objects
 - runs execute on multiapps (including FullSolveMultiApp)
 - outputs on initial (initial condition)

permcody committed:
Check input does too much

This PR moves the --check-input functionality to the MooseApp::run() method where it makes more sense

refs #4438

moosebuild commented:
Job `App tests:linux-gnu` on c17715c : invalidated by @permcody

BISON PR Merged

andrsd commented:
The code change looks good to me.

The fail in debug modules is because of a timeout.

And BISON and Rattlesnake segfaults on one test each where they check syntax. But maybe the apps need to be patched to work with this change...

permcody commented:
Yeah, I tried to delete the CLI overrides for resizing the mesh. I guess that's a bad idea for some of these bigger meshes. However messing with the problem with these check syntax tests is exactly why my test failed. I'd rather not do that because it kind of defeats the purpose of checking the syntax when we blatantly ignore part of it. It looks like these really big meshes read in relatively fast for optimized mode, just not debug. So we have two choices, either run the check syntax in "!DBG", or put the mesh resizing back in for some of these larger tests. Thoughts?

andrsd commented:
A lot of the mesh size changes is properly documented in the `tests` files, so I would keep that in. Running in debug is a good thing.

permcody committed:
Switch AppFactory to shared pointers

This patch switches the AppFactory over to shared pointers similarly to the
way the Factory works. This is to avoid deletion problems encountered when
we use the newly placed --check-input flag. We'll maintain the dumb pointer
version for awhile until all main.C applications can be patched

permcody committed:
Adjust a few of the check_input checks

This is being down now because check_input is slightly cheaper. Also
many developers run in modes other than OPT.

refs #4438

permcody commented:
@andrsd - Sorry this PR has turned into a big set of commits. I did some smart pointer stuff to clean up destructors that started failing with the simple move of this logic. 

It's ready for review but we'll have to wait for the rattlesnake submodule update in Mammoth before this can be merged.

andrsd commented:
No problem, that happens.

moosebuild commented:
Job `App tests:linux-gnu` on 948fd4e : invalidated by @permcody

Mammoth submodule updated

dschwen commented:
Noooooooooo

On Tue, Apr 18, 2017, 3:02 PM Cody Permann <notifications@github.com> wrote:

> Merged #8948 <https://github.com/idaholab/moose/pull/8948>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/8948#event-1047438599>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAMWPjAE9Z2OCAWL7qxvGefqvvYVFqd8ks5rxSTZgaJpZM4M_ogW>
> .
>

dschwen commented:
This weakens the test considerably.

permcody commented:
I disagree. It makes them more accurate to the name. If we want a small
test mode that runs up through initial conditions, I'm happy to entertain
that separately. This option was originally added to catch changes in
syntax which are prevalent in our workflow. If you make it past syntax
parsing, you are generally good to go.
On Tue, Apr 18, 2017 at 7:48 PM Daniel Schwen <notifications@github.com>
wrote:

> This weakens the test considerably.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/8948#issuecomment-295038442>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AC5XIMO6NZV4UWdzi1z-rPWZpIXSKSdAks5rxWfagaJpZM4M_ogW>
> .
>
